pkg/network/multus/status_test.go

```diff
@@ -33,72 +32,133 @@ import (
+			),
+			Entry("when network status annotation is empty",
+				map[string]string{networkv1.NetworkStatusAnnot: ""},
+			),
+			Entry("when network status annotation is illegal",
+				map[string]string{networkv1.NetworkStatusAnnot: "not a valid JSON array"},
+			),
+		)
 
-		expectedResult := map[string]networkv1.NetworkStatus{
-			"pod7e0055a6880": {
-				Name:      "meganet",
-				Interface: "pod7e0055a6880",
-				Mac:       "8a:37:d9:e7:0f:18",
-				Default:   false,
-			},
-		}
+		It("Should return a valid network status slice", func() {
+			const multusNetworkStatusWithPrimaryAndSecondaryNets = `[` +
+				`{"name":"k8s-pod-network","ips":["10.244.196.146","fd10:244::c491"],"default":true,"dns":{}},` +
+				`{"name":"meganet","interface":"pod7e0055a6880","mac":"8a:37:d9:e7:0f:18","dns":{}}` +
+				`]`
 
-		Expect(result).To(Equal(expectedResult))
+			annotations := map[string]string{networkv1.NetworkStatusAnnot: multusNetworkStatusWithPrimaryAndSecondaryNets}
+
+			expectedResult := []networkv1.NetworkStatus{
+				{
+					Name:    "k8s-pod-network",
+					IPs:     []string{"10.244.196.146", "fd10:244::c491"},
+					Default: true,
+					DNS:     networkv1.DNS{},
+				},
+				{
+					Name:      "meganet",
+					Interface: "pod7e0055a6880",
+					Mac:       "8a:37:d9:e7:0f:18",
+					Default:   false,
+					DNS:       networkv1.DNS{},
+				},
+			}
+
+			Expect(multus.NetworkStatusesFromPod(newStubPod(annotations))).To(Equal(expectedResult))
+		})
 	})
 
-	It("Should return a map of pod interface name to network status when interface name is ordinal", func() {
+	Context("LookupPodPrimaryIfaceName", func() {
 		const (
-			multusOrdinalNetworksAnnotation                       = `[{"name":"meganet","namespace":"default","interface":"net1"}]`
-			multusNetworkStatusWithPrimaryAndOrdinalSecondaryNets = `[` +
-				`{"name":"k8s-pod-network","ips":["10.244.196.146","fd10:244::c491"],"default":true,"dns":{}},` +
-				`{"name":"meganet","interface":"net1","mac":"8a:37:d9:e7:0f:18","dns":{}}` +
-				`]`
+			defaultPrimaryPodIfaceName = "eth0"
+			customPrimaryPodIfaceName  = "custom-iface"
 		)
 
-		annotations := map[string]string{
-			networkv1.NetworkAttachmentAnnot: multusOrdinalNetworksAnnotation,
-			networkv1.NetworkStatusAnnot:     multusNetworkStatusWithPrimaryAndOrdinalSecondaryNets,
-		}
-
-		result := multus.NonDefaultNetworkStatusIndexedByIfaceName(newStubPod(annotations))
-
-		expectedResult := map[string]networkv1.NetworkStatus{
-			"net1": {
-				Name:      "meganet",
-				Interface: "net1",
-				Mac:       "8a:37:d9:e7:0f:18",
-				Default:   false,
-			},
-		}
+		DescribeTable("should return empty string", func(networkStatuses []networkv1.NetworkStatus) {
+			Expect(multus.LookupPodPrimaryIfaceName(networkStatuses)).To(BeEmpty())
+		},
+			Entry("when network status is nil", nil),
+			Entry("when network status is empty", []networkv1.NetworkStatus{}),
+			Entry("when network status does not report interface name",
```

Comment:
should we add an entry that is "legit" (i.e. features the interface name) but is *not* the default net ?

Reply:
Could you please give an example of such a scenario? (preferably the JSON value)

Reply:
Done.